pyproject.toml

```diff
@@ -31,12 +30,12 @@ classifiers = [
 ]
 
 [tool.poetry.dependencies]
-python   = "^3.9"
+python   = "^3.10"
```

Comment:
NI's python library policy is to always support all non-EOL python versions, which includes 3.9 until October 31 of this year.

What we actually want in this case is to conditionally require the higher version when installing to python 3.10 or newer. Poetry [supports this](https://python-poetry.org/docs/dependency-specification/#multiple-constraints-dependencies).

```toml
uplink = [
  { version = "^0.10.0", extras = ["pydantic"], python_version = ">=3.10" },
  { version = "^0.9.1", python_version = ">=3.9,<3.10" }
]
```